### PR TITLE
Implement the spec

### PR DESCRIPTION
**Pull request by paf31**
@jdegoes 

I've added tests for the vast majority of the implemented methods, but I had to omit a few.

The example contains a few hand-written tests for some common tasks, mainly event handlers, and a lot of generated tests.

I updated the license file to reflect the fact that I used the TypeScript definition file from DefinitelyTyped as part of my code generation process.

Please let me know what you think.

Thanks.

### CODE REVIEW

File: src/Ace/Internal.purs [hunk 1]
```diff
@@ -0,0 +1,19 @@
+module Ace.Internal
```
Comment (jdegoes) [inline src/Ace/Internal.purs]: Would this be a good candidate for a foreign lib (purescript-nullable or something)?

Comment (paf31) [inline src/Ace/Internal.purs]: Yes, I can put that together.

**Comment by jdegoes** [comment]
Just two comments overall:
1. Can you setup a Travis build file for the project?
2. Can you extract out purescript-nullable and have this project depend on that one.

Otherwise, looks good to go!

By the way, I don't think it's necessary to modify the license, since no portions of the source code were used or copied from the Typescript definition files (rather, the definition files were used as a reference input data set to a 3rd party code generator which then produced de nova Purescript files, which include neither the Typescript nor the Typescript definitions).

**Comment by paf31** [comment]
Here you go: https://travis-ci.org/paf31/purescript-ace/builds/48808533

**Comment by jdegoes** [comment]
:+1: Thanks for your work on this! Bug Jeff about the bounty. :)

**Comment by paf31** [comment]
Will do, thanks! :)

You might need to turn on Travis separately for this repo, I'm not sure.

**Pull request merged**